rosserial_client/src/rosserial_client/make_library.py

```diff
@@ -550,12 +559,8 @@ def rosserial_generate(rospack, path, mapping):
     # gimme messages
     failed = []
     for p in sorted(rospack.list()):
-        try:
-            MakeLibrary(p, path, rospack)
-        except Exception as e:
-            failed.append(p + " ("+str(e)+")")
-            print('[%s]: Unable to build messages: %s\n' % (p, str(e)))
-            print(traceback.format_exc())
+        rosserial_generate_package(rospack, p, path, mapping)
```

Comment:
When you extracted the `rosserial_generate_package` function, you removed the lines that print the exception and the traceback as well as the line that appends to the list `failed` While I personally don't mind it that much, I can't see this being accepted if we ever open a pull request. I'm OK with merging it in the current state and worrying about polishing it later, if you are too.